### PR TITLE
Add `log_device_info` to Trainer

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Add `log_device_info` to Trainer

## Before submitting
- [] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [] Did you make sure to update the documentation with your changes? (if necessary)
- [] Did you write any new necessary tests? (not for typos and docs)
- [] Did you verify new and existing tests pass locally with your changes?
- [] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -1213,3 +1223,30 @@ def __setup_profiler(self) -> None:
...
         self.profiler.setup(stage=self.state.fn._setup_fn, local_rank=local_rank, log_dir=self.log_dir)
+
+    def log_device_info(self) -> None:
+        rank_zero_info(f'GPU available: {torch.cuda.is_available()}, used: {self._device_type == DeviceType.GPU}')
+
+        num_tpu_cores = self.tpu_cores if self.tpu_cores is not None else 0
+        rank_zero_info(f'TPU available: {_TPU_AVAILABLE}, using: {num_tpu_cores} TPU cores')
+
+        num_ipus = self.ipus if self.ipus is not None else 0
+        rank_zero_info(f'IPU available: {_IPU_AVAILABLE}, using: {num_ipus} IPUs')
```
Comment (carmocca) [inline pytorch_lightning/trainer/trainer.py:1234]: Should we print these only if available?

**Comment by carmocca** [comment]
What's the motivation behind this?

File: pytorch_lightning/trainer/trainer.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/trainer/trainer.py:1234]: no, always. 
always good to know what machine we are running on :)

Comment (kaushikb11) [reply pytorch_lightning/trainer/trainer.py:1234]: Indeed.

**Review by awaelchli — APPROVED** [review]
LGTM but I think we should consider making it a protected api, since Trainer doesn't log anything normally

**Comment by kaushikb11** [comment]
> LGTM but I think we should consider making it a protected api, since Trainer doesn't log anything normally

Updated!

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**